### PR TITLE
ci: add link checker and fix links

### PR DESCRIPTION
**Pull request by katexochen**
Contributing to https://github.com/confidential-containers/confidential-containers/issues/83

### CODE REVIEW

File: docs/shim-rune.md [hunk 1]
```diff
@@ -134,7 +134,7 @@ uses the following methods to run the app enclave container.
 ### Occlum
 
 Refer to the occlum guide of
-[Runtime boot pre-generated UnionFS image](https://github.com/qzheng527/occlum/blob/combine_jsons/demos/runtime_boot).
+[Runtime boot pre-generated UnionFS image](https://github.com/occlum/occlum/tree/master/demos/runtime_boot).
```
Comment (Xynnn007) [inline docs/shim-rune.md:137]: Please double check this link @qzheng527

File: docs/shim-rune.md [hunk 2]
```diff
@@ -157,7 +157,7 @@ One host only needs a pre-installed occlum OCI bundle for the boot instance. The
 
 The pre-generated boot template instance looks like an OCI bundle rootfs on the host side.
 It contains the
-[occlum boot template instance](https://github.com/qzheng527/occlum/blob/combine_jsons/demos/runtime_boot/README.md#build-a-boot-template-occlum-instance).
+[occlum boot template instance](https://github.com/occlum/occlum/blob/master/demos/runtime_boot/README.md#build-a-boot-template-occlum-instance).
```
Comment (Xynnn007) [inline docs/shim-rune.md:160]: cc @qzheng527

**Review by Xynnn007 — COMMENTED** [review]
Wow. Looks like a cool thing! Thanks @katexochen

**Review by dcmiddle — APPROVED** [review]
This looks very helpful! Thanks @katexochen

File: docs/shim-rune.md (hunk 1 above)
Comment (mythi) [reply docs/shim-rune.md:137]: this is correct but to make it match with the code, we need #180

File: .github/workflows/links.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  checklinks:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (mythi) [inline .github/workflows/links.yml:15]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/links.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  checklinks:
+    runs-on: ubuntu-latest
```
Comment (mythi) [inline .github/workflows/links.yml:12]:
```suggestion
    runs-on: ubuntu-22.04
```

File: .lycheeignore [hunk 5]
```diff
@@ -1 +0,0 @@
-/vendor/
```
Comment (mythi) [inline .lycheeignore]: please squash the commits so that `.lycheeignore` is not added/removed

**Review by mythi — COMMENTED** [review]
looks nice! a couple of minor suggestions

**Review by mythi — APPROVED** [review]
Thanks! I'm fine merging this for the release even if we don't have #180. @Xynnn007 WDYT?

**Review by Xynnn007 — APPROVED** [review]
> Thanks! I'm fine merging this for the release even if we don't have #180. @Xynnn007 WDYT?

Agreed. Thanks for this @katexochen !

**Pull request merged**